src/redis-check-aof.c

```diff
@@ -294,32 +301,34 @@ int processAnnotations(FILE *fp, char *filename, int last_file) {
     epos = ftello(fp);
     if (fgets(buf, sizeof(buf), fp) == NULL) {
         printf("Failed to read annotations from AOF %s, aborting...\n", filename);
-        fclose(fp);
         exit(1);
     }
 
     if (to_timestamp && strncmp(buf, "#TS:", 4) == 0) {
-        time_t ts = strtol(buf+4, NULL, 10);
+        char *endptr;
+        errno = 0;
+        time_t ts = strtol(buf+4, &endptr, 10);
+        if (errno != 0) {
```

Comment:
What about checking endptr?

Reply:
Since the buf here is not a c string terminated with '\0', I am not sure if we are going to use `*endptr != '\r' `to judge.

Reply:
Added